### ISSUE TITLE
Mocha stops before finishing all tests

### ISSUE THREAD

**Issue by ArcticZeroo** [mochajs/mocha#3396]
### Prerequisites

- [x] Checked that your issue hasn't already been filed by cross-referencing [issues with the `faq` label](https://github.com/mochajs/mocha/issues?utf8=%E2%9C%93&q=is%3Aissue%20label%3Afaq%20)
- [x] Checked next-gen ES issues and syntax problems by using the same environment and/or transpiler configuration without Mocha to ensure it isn't just a feature that actually isn't supported in the environment in question or a bug in your code.
- [x] 'Smoke tested' the code to be tested by running it outside the real test suite to get a better sense of whether the problem is in the code under test, your usage of Mocha, or Mocha itself
- [x] Ensured that there is no discrepancy between the locally and globally installed versions of Mocha. You can find them with: `node node_modules/.bin/mocha --version`(Local) and `mocha --version`(Global). We recommend avoiding the use of globally installed Mocha.

### Description

Mocha isn't finishing all my tests! 

This is the only file I currently have to run my tests: https://github.com/ArcticZeroo/carabiner/blob/master/test/test.js

As you can see, there are 7 "high level" categories described within "Carabiner". However, after the 5th (Client), mocha stops. 

I thought that perhaps something may have gone wrong within the prior tests that would lock up mocha or something, or that perhaps mocha had a limit to the number of descriptions. So I tried to comment out other tests, including "Client", everything before "Client", etc and it's still stopping at Client. 

And I have absolutely no idea why!

The version of mocha I have on my machine is 3.5, which I am aware is two major versions out of date. However, travis' version is 5.2.0 (latest I presume, since the command run is just npm i -g mocha) and it's running into the same issue.

Here is the travis build for it: https://travis-ci.org/ArcticZeroo/carabiner

### Steps to Reproduce

- Clone https://github.com/ArcticZeroo/carabiner
- Run "npm i"
- Get a slack token from somewhere. Doesn't matter where, because if you are able to reproduce my issue, you'll never even hit the testing environment where the contents of the organization actually matters.
- Set this token to the env variable SLACK_TOKEN (set SLACK_TOKEN=yourtoken) 
- Run "npm test" (which just runs "mocha")
- See that all tests in Carabiner up to client pass, and then mocha seems to think it's finished

**Expected behavior:** Mocha tests all 7 high level "descriptions" inside Carabiner

**Actual behavior:** Mocha tests the first 5

**Reproduces how often:** 100%

### Versions

I don't think version is relevant here. As I said, it occurs on 5.2 and 3.5, along with node 7 and 8, on Windows and whatever Travis uses (some type of linux?)

### Additional

Here's a screenshot of the current output on my machine, and it does exit once this is done
http://frozor.io/up/xLkVtV0.png

Ignore the experimental warning, that is a super recent thing that I am pretty sure is not related.

**Comment by ArcticZeroo** [comment]
Further weirdness:

I was looking for why my Travis build was halting for 10m, and saw the FAQ thing about leaving sockets open. Which made me realize I was indeed leaving Slack RTM websockets open, and so I went to track down the cause.  I added debug to print a stacktrace each time an RTM socket was opened, and it appears that the "test slack organization" function is actually called... but mocha never seems to complete it? I'm not sure what's up.
